src/core/or/circuitpadding.c

```diff
@@ -539,6 +539,11 @@ circpad_machine_sample_delay(circpad_machine_runtime_t *mi)
     histogram_total_tokens = state->histogram_total_tokens;
   }
 
+  /* If we are out of tokens, don't schedule padding. */
```

Comment:
What triggered this condition? It seems OK as a safety check, but shouldn't our machines transition to the END state instead of relying on this?

Reply:
Hmm, this triggers for the origin-side rendezvous machine right now, because of this transition:
```
  machine->states[CIRCPAD_STATE_BURST].
    next_state[CIRCPAD_EVENT_PADDING_SENT] = CIRCPAD_STATE_BURST;
```
and the fact that the code `circpad_send_padding_cell_for_callback()` first does the `circpad_cell_event_padding_sent(circ);`  transition and attempts to send padding with zero tokens, and then calls `check_machine_token_supply()` to potentially bring the machine to `END`.

The BURST->BURST transition above is not necessary because the machine only has one token, but in previous iterations of the code it had more than one token and that's why I had that transition.

What do you think?